### DISCUSSION TITLE
Feature Request: mark partial chunks of code as approved in a Pull Request

### DISCUSSION THREAD

**Discussion by chalofa** [community/community#22041] [New to GitHub]
<p>When I’m reviewing big Pull Request with a lot of changes, it would be pretty useful if there were a way to " <strong>mark this code chunk as reviewed</strong>", and collapse/hide it.</p>
<p>This way, I could review a few lines of code, “hide” the ones I think work great, and focus on the lines that could be getting more attention to me… I think this could work similar or tied to the “Start review” process for comments…</p>

**Comment by BelHFL** [comment]
<p>Hi <a class="mention" href="/chalofa">@chalofa</a> ,</p>
<p>Thanks for this feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Cheers!</p>

**Comment by mariozaizar** [comment]
<p>I would love to be able to mark files as reviewed and expect them to be collapsed in the UI. Good idea!</p>

**Comment by chalofa** [comment]
<p>Thanks!</p>
<p>There’s even a meme for this:</p>
<p><img src="https://user-images.githubusercontent.com/217029/181068444-9a21be12-9ed3-42cb-95d3-594e75d6192c.png" alt="meme.png" data-base62-sha1="gSuCElfuCyZtExc3pzxfBS2ClcX" width="498" height="326"></p>

**Comment by chalofa** [comment, marked as answer]
<p>Today, I see a <code>Viewed</code> link in each file changed, so, thanks!</p>
<p>Thanks!</p>
<p><img src="https://user-images.githubusercontent.com/217029/181068445-807ce020-140e-4e82-b71f-38d13dafafca.png" alt="feature.png" data-base62-sha1="3Zy1NBnM2rj02TVcMxWZrPmcpgd" width="366" height="314"></p>

**Comment by axilis-marko** [comment]
<p>Will this option ever be introduced for git <code>hunks</code>? For big files with a lot of changes, it would help keep a PR readable.</p>

**Comment by tolomea** [comment]
This feature is great for small files, but for bigger files it really needs a way to mark parts of the file as viewed. And chunks won't really cut it. Classes and functions would be better. Or maybe a hybrid where you split chunks at global class / function boundary.
For context the review I'm doing right now has a new Python file that is 300 lines long most of which is in three classes. I'd really like to be able to tick those off one by one before ticking off the rest of the file.
